azure-iot-device/samples/advanced-hub-scenarios/register_symmetric_key.py

```diff
@@ -0,0 +1,53 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+
+# This is for illustration purposes only. The sample will not work currently.
+
+import os
+import logging
+import asyncio
+from azure.iot.device import SymmetricKeySecurityClient
+from azure.iot.device.aio import SymmetricKeyProvisioningDeviceClient
+
+
+logging.basicConfig(level=logging.DEBUG)
+
+provisioning_host = os.getenv("PROVISIONING_HOST")
+id_scope = os.getenv("PROVISIONING_IDSCOPE")
+registration_id = os.getenv("PROVISIONING_REGISTRATION_ID")
+symmetric_key = os.getenv("PROVISIONING_SYMMETRIC_KEY")
+
+
+async def register_device():
+    symmetric_key_security_client = SymmetricKeySecurityClient(
+        provisioning_host, registration_id, symmetric_key, id_scope
+    )
+    provisioning_device_client = SymmetricKeyProvisioningDeviceClient.create_from_security_client(
+        symmetric_key_security_client, "mqtt"
+    )
+
+    await provisioning_device_client.register()
+
+
+async def main():
+    task1 = asyncio.create_task(register_device())
+    await task1
+
+    # If using Python 3.6 or below, use the following code instead of asyncio.run(main()):
+    # await register_device()
```

Comment:
I don't think this is true

Reply:
yes it will. The upper one with the task is better convention. Specially when there are multiple tasks there is some difference versus the lower one. So preferred way is upper whenever possible.

---
In reply to: [289560351](https://github.com/Azure/azure-iot-sdk-python-preview/pull/98#discussion_r289560351) [](ancestors = 289560351)